generator/templates/model.tgo

```diff
@@ -41,7 +41,10 @@ func (r *{{.Name}}) SetRelationship(field string, rel interface{}) error {
                 if !ok {
                         return fmt.Errorf("kallax: record of type %t can't be assigned to relationship {{.Name}}", rel)
                 }
-                r.{{.Name}} = {{if or (not .IsPtr) .IsOneToManyRelationship}}*{{end}}val
+                {{if .IsPtr}}if !val.Model.ID.IsEmpty() {
+                {{end}}r.{{.Name}} = {{if or (not .IsPtr) .IsOneToManyRelationship}}*{{end}}val{{if .IsPtr}}
+                }
+                {{end}}
```

Comment:
`text/template` does weird things with blank on the `{{stuff}}`, so not much I can do on that side :(

But splitting this in two branches is nice, i'll change this asap